x-pack/plugins/watcher/public/sections/watch_edit/watch_edit_actions.ts

```diff
@@ -75,21 +75,30 @@ export async function saveWatch(watch: BaseWatch, licenseService: any) {
 }
 
 export async function validateActionsAndSaveWatch(watch: BaseWatch, licenseService: any) {
-  const { warning } = watch.validate();
```

Comment:
Do you know what this used to do? I'm just curious about how the UX has changed with the removal of this logic.

Reply:
The original code displayed this modal if the `to` field was missing from a slack action.

<img width="811" alt="Screen Shot 2019-04-11 at 8 48 33 AM" src="https://user-images.githubusercontent.com/5226211/56746954-2441a200-674b-11e9-8da8-e73d3de607ec.png">

However, when setting up an integration with Slack, it seems it's required to provide a "to" value.

<img width="1056" alt="Screen Shot 2019-04-25 at 11 17 25 AM" src="https://user-images.githubusercontent.com/5226211/56747317-bba6f500-674b-11e9-8118-2edf335a157f.png">

If an option is not selected, the save button is disabled.

<img width="331" alt="Screen Shot 2019-04-25 at 11 15 46 AM" src="https://user-images.githubusercontent.com/5226211/56747316-bb0e5e80-674b-11e9-876c-405f4e5bb47f.png">

The value can be overridden in ES, but if it's not defined, the notification will go to the channel specified in Slack.

Also see: https://github.com/elastic/kibana/issues/18531#issuecomment-384127356

@sebelga I know I talked to you about this originally. Does the above make sense? Or I am missing another case?

Reply:
Cool! If I understand correctly, we originally thought that a Slack watch saved without a `to` field either on the watch itself or in the ES settings would be essentially useless, because it wouldn't end up doing anything when triggered. However, we've learned that Slack allows you to use a Slack watch even if there is no `to` field specified on our end, rendering this warning moot/incorrect.

Reply:
I am not sure this is correct. From the comment

> The way I setup my Slack Webhook for this watch, it has a default "to" on the slack side.

So a "to" _is_ required. The thing is, the user can set it multiple ways. The warning that was put in place is: "we can't validate that you have set properly a "to" field, be aware that the watch might not work if you haven't specified it outside our UI".

This validation was useful if the user enters "Create advanced watch", add a JSON payload of the watch with a Slack action, but without a "to" field.

Reply:
@cjcenizal that's correct.

@sebelga As far as I can tell, you would have to **at minimum** set a "to" via Slack. So while you could override it via ES, I think you would **always** have to set it in Slack (see screenshots above) in order to generate the webhook url. Does that make sense?

Basing this on the docs here and my own setup experience: https://www.elastic.co/guide/en/elastic-stack-overview/7.0/actions-slack.html#configuring-slack

Also, note in the docs:

> You can no longer configure Slack accounts using elasticsearch.yml settings. Please use Elasticsearch’s secure keystore method instead.